File: src/2574.left-and-right-sum-differences.2714/solution.ts
function leftRigthDifference (nums: number[]): number[] {
    const rightSum = new Array<number>(nums.length);
    rightSum[rightSum.length - 1] = 0;
    for (let i = nums.length - 2; i > -1; i--) {
        rightSum[i] = rightSum[i + 1] + nums[i + 1];
    }
    const result = new Array<number>(nums.length);
    let leftSum = 0;
    for (let i = 0; i < nums.length; i++) {
        result[i] = Math.abs(rightSum[i] - leftSum);
        leftSum += nums[i];
    }

    return result;
}

File: src/2574.left-and-right-sum-differences.2714/solution.java
class Solution {
    public int[] leftRightDifference(int[] nums) {
        int[] rights = new int[nums.length];
        int right = 0;
        for (int i = nums.length - 1; i > -1; i--) {
            rights[i] = right;
            right += nums[i];
        }
        int[] result = new int[nums.length];
        int left = 0;
        for (int i = 0; i < nums.length; i++) {
            result[i] = Math.abs(left - rights[i]);
            left += nums[i];
        }
        return result;
    }
}